doc/SAI-Proposal-Generic-Resource-Monitoring.md

```diff
@@ -0,0 +1,51 @@
+Switch Abstraction Interface Change Proposal
```

Comment:
Why cant we just use following attributes from ```saiswitch.h``` 
```

    /**
     * @brief Available IPv4 routes
     *
     * @type sai_uint32_t
     * @flags READ_ONLY
     */
    SAI_SWITCH_ATTR_AVAILABLE_IPV4_ROUTE_ENTRY,

    /**
     * @brief Available IPv6 routes
     *
     * @type sai_uint32_t
     * @flags READ_ONLY
     */
    SAI_SWITCH_ATTR_AVAILABLE_IPV6_ROUTE_ENTRY,

    /**
     * @brief Available IPv4 Nexthop entries
     *
     * @type sai_uint32_t
     * @flags READ_ONLY
     */
    SAI_SWITCH_ATTR_AVAILABLE_IPV4_NEXTHOP_ENTRY,
```

Reply:
These attributes are hard coded. The purpose of this proposal is to introduce a generic API that will allow the query of any type of resource without introducing more attributes to the switch object.

Reply:
So you intend to deprecate these attributes in switch once this proposal is approved.

Reply:
Yes, that's correct.